example-watcher/index.js

```diff
@@ -0,0 +1,29 @@
+const { notificationTypes } = require("./index.json");
+
+async function watch({ latestData, updateData, submitNotifications, libs }) {
+  const notifications = [];
+
+  // you can use libs to obtain and parse info
+  // eslint-disable-next-line no-console
+  console.log("Available libs:", Object.keys(libs));
+
+  // compare with latestData to generate notifications
+  const newData = {
+    a: 2
+  };
+
+  if (latestData.a !== newData.a) {
+    notifications.push({
+      key: notificationTypes.newValue.key,
+      message: `a has a new value: ${newData.a}`
+    });
+  }
+
+  // save new data with updateData for next watch
+  updateData(newData);
```

Comment:
Probably add a `catch` here. We should send the notification only if the data was saved properly. Or else, if the data saving fails, then the user might get the same notification 2 times.

Reply:
changed to return a `{ snapshot, notifications }` object, this leaves the `catch` logic you mentioned in the hands of the server